scripts/Muon/GUI/FrequencyDomainAnalysis/FFT/fft_model.py

```diff
@@ -0,0 +1,155 @@
+    This keeps the main FFT class simple.
+    """
+
+    def __init__(self, FFT):
+        self.name = "FFT"
+        self.model = FFT
+
+    def cancel(self):
+        self.model.cancel()
+
+    def loadData(self, inputs):
+        """
+        store the data in the wrapper for later
+        """
+        if "phaseTable" in inputs:
+            self.phaseTable = inputs["phaseTable"]
+        else:
+            self.phaseTable = None
+        if "preRe" in inputs:
+            self.preRe = inputs["preRe"]
+        else:
+            self.preRe = None
+        if "preIm" in inputs:
+            self.preIm = inputs["preIm"]
+        else:
+            self.preIm = None
+        if "FFT" in inputs:
+            self.FFT = inputs["FFT"]
+        else:
+            self.FFT = None
+        self.model.setRun(inputs["Run"])
+
+    def execute(self):
+        """
+        runs the relevant parts of the FFT and the preprocessing
+        """
+        if self.phaseTable is not None:
+            if self.phaseTable["newTable"]:
+                self.model.makePhaseQuadTable(self.phaseTable)
+            self.model.PhaseQuad()
+
+        if self.preRe is not None:
+            self.model.preAlg(self.preRe)
+
+        if self.preIm is not None:
+            self.model.preAlg(self.preIm)
+
+        if self.FFT is not None:
+            self.model.FFTAlg(self.FFT)
+
+    def output(self):
+        return
+
+
+class FFTModel(object):
+
+    """
+    A simple class which executes
+    the relevant algorithms for
+    the analysis.
+    """
+
+    def __init__(self):
+        self.name = "FFT"
+        self.alg = None
+
+    def cancel(self):
+        if self.alg is not None:
+            self.alg.cancel()
+
+    def setRun(self, run):
+        self.runName = run
+
+    def preAlg(self, preInputs):
+        """
+        PaddingAndApodization alg on the data
+        """
+        self.alg = mantid.AlgorithmManager.create("PaddingAndApodization")
+        self.alg.initialize()
+        self.alg.setAlwaysStoreInADS(False)
+        for name, value in iteritems(preInputs):
+            self.alg.setProperty(name, value)
+        self.alg.execute()
+        mantid.AnalysisDataService.addOrReplace(
+            preInputs["OutputWorkspace"],
+            self.alg.getProperty("OutputWorkspace").value)
+        self.alg = None
+
+    def FFTAlg(self, FFTInputs):
+        """
+        Use the FFT alg
+        """
+        self.alg = mantid.AlgorithmManager.create("FFT")
+        self.alg.initialize()
+        self.alg.setAlwaysStoreInADS(False)
+        for name, value in iteritems(FFTInputs):
+            self.alg.setProperty(name, value)
+        self.alg.execute()
+        mantid.AnalysisDataService.addOrReplace(
+            FFTInputs["OutputWorkspace"],
+            self.alg.getProperty("OutputWorkspace").value)
+
+        ws = self.alg.getPropertyValue("OutputWorkspace")
+        group = mantid.AnalysisDataService.retrieve(self.runName)
```

Comment:
Is it worth checking that the workspace exists when querying the ADS, just in case something strange happened or the naming changed from the GUI?

Reply:
the check is done in the view

Reply:
Is the possiblity that `retrieve` could raise an exception if the workspace had disappeared, by being deleted/renamed by another task, taken into account somewhere?

Reply:
if any exceptions are created it is caught in the thread model, we could add a check if its not none

Reply:
:+1:

Reply:
I made a mistake in the presenter for FFT, as it did not set up the thread model correctly. Fixing that means if a workspaces is missing a nice pop up appears with an error message (no other code changes needed)